### PR TITLE
LOG-4231: CLO operator support for optional OTEL data model

### PR DESCRIPTION
**Pull request by Clee2691**
## Description
This PR is a continuation of the effort by Jatin Suri to support transforming the log record formats to the OpenTelemetry semantic.

This feature is:
- Locked behind a feature-gate
- Applies only to `application` logs
- Can only be defined for `HTTP` outputs under the `HTTP` output spec

## Field Changes:
| Original| New	|
|---|---|
| @timestamp |	timeUnixNano |
| level	|severityText |
| N/A	|	severityNumber |
| file	|resources.logs.file.path |
| hostname	|resources.host.name |
| kubernetes.container_name	|resources.container.name |
| kubernetes.container_id	|resources.container.id |
| kubernetes.container_image	|resources.container.image.name |
| N/A	|resources.container.image.tag |
| kubernetes.pod_name	|resources.k8s.pod.name |
| kubernetes.pod_id	|resources.k8s.pod.uid |
| kubernetes.pod_ip	|resources.k8s.pod.ip |
| kubernetes.pod_owner	|resources.k8s.pod.owner |
| kubernetes.annotations	|resources.k8s.pod.annotations |
| kubernetes.labels	|resources.k8s.pod.labels |
| kubernetes.namespace_id	|resources.k8s.namespace.id |
| kubernetes.namespace_labels."kubernetes.io/metadata.name"	|resources.k8s.namespace.name |
| kubernetes.namespace_labels	|resources.k8s.namespace.labels |
| log_type	|resources.attributes.log_type |

/cc @alanconway @cahartma
/assign @jcantrill 

### Links
- Based on PR: https://github.com/openshift/cluster-logging-operator/pull/2142
- JIRA: https://issues.redhat.com/browse/LOG-4231
- Enhancement Proposal: https://github.com/openshift/enhancements/pull/1442

### CODE REVIEW

File: internal/generator/vector/output/http/http.go
Comment (alanconway) [inline internal/generator/vector/output/http/http.go:138]: Need nil check in case o is not a HTTP output.

**Review by alanconway — APPROVED** [review]
This will work, restricts OTEL to the HTTP output which we will want to open up in the long run, but it's a good conservative choice for the preview given unresolved headaches.
Ultimately I want to move OTEL to a "filter" like the API audit stuff, but we should go ahead with this approach as the first cut since we may not have time to clean up all the loose ends after I merge the audit filter.

**Comment by alanconway** [comment]
See comment about missing nil check - then I'll LGTM

File: internal/generator/vector/normalize/schema/otel/transform.go
Comment (jcantrill) [inline internal/generator/vector/normalize/schema/otel/transform.go:30]: severity is not the same as facility (e.g. https://success.trendmicro.com/dcx/s/solution/TP000086250?language=en_US) .  Regardless, I think our fluent code has 0-9 but I believe 0-7 have always been the only ones available

File: internal/generator/vector/normalize/schema/otel/transform.go
Comment (jcantrill) [inline internal/generator/vector/normalize/schema/otel/transform.go:52]: if this is missing will this return empty string or null?

File: internal/generator/vector/output/http/http.go
Comment (jcantrill) [inline internal/generator/vector/output/http/http.go:138]: More nit then anything, I defer to you if you wish to change.  Options#Has(key) exists as a method and we explicitly give a value to Options meaning we don't care about the actual value.  It may read cleaner to adjust

File: test/helpers/types/types.go
Comment (jcantrill) [inline test/helpers/types/types.go:33]: Do we need to add this into a different top level log type (e.g. OTELContainerLog, OTELLog)? I'm assuming OTEL format will likely be agnostic of Container versus other.  I'm thinking ahead of when we no longer normalize VIAQ at all and seems it would be more appropriate to have start the different normalization down their own path.  

Consider moving all "otel" struct to a separate file

**Review by jcantrill — CHANGES_REQUESTED** [review]

File: internal/generator/vector/normalize/schema/otel/transform.go
Comment (Clee2691) [reply internal/generator/vector/normalize/schema/otel/transform.go:52]: It returns an empty string

File: test/helpers/types/types.go
Comment (Clee2691) [reply test/helpers/types/types.go]: Move to `test/helpers/schema/otel/types.go`.

**Review by jcantrill — APPROVED** [review]
/lgtm

We could actually do normalization tests in the same way as the others with templates.  Consider doing that in future so we can test "all" fields in lieu of verifying individually

**Pull request merged**
